tests/4.5/target_enter_exit_data/test_target_enter_exit_data_devices.c

```diff
@@ -1,45 +1,55 @@
+//===--- test_target_enter_exit_data_devices.c---------------------------------------===//
+//
+//  OpenMP API Version 4.5 Nov 2015
+//
+//  This test checks for data mapping on multiple devices when using the target
+//  data map directive. The test uses omp_get_num_devices to confirm the number 
+//  of total devices available. There are two separate functions, and both make sure 
+//  that data mapping is happening on each available device. The first function does 
+//  this through the use of omp_set_default_device, while the second function explicity 
+//  uses the device() clause to control which device is being utilized.
+//
+//  Since OpenMP 4.5 does not have an API call to obtain the current device, 
+//  this test does not guarantee that the execution devices are different. 
+//
+////===------------------------------------------------------------------------------===/
+
 #include <assert.h>
 #include <omp.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include "ompvv.h"
 
 #define N 1000
 
 int test_set_default_dev() {
 
-  printf("test_set_default_dev\n");
+  OMPVV_INFOMSG("test_set_default_dev");
 
   // Get number of devices
   int num_dev = omp_get_num_devices();
-  printf("num_devices: %d\n", num_dev);
+  OMPVV_INFOMSG("num_devices: %d", num_dev);
 
   int def_dev = omp_get_default_device();
```

Comment:
Although I don't find a relationship between num_devices and default device no, I wonder if the device no is between 1 and num_dev.

Reply:
Check what value default is ->

Reply:
On Fatnode -> num_dev = 2, default_dev = 0, and initial device = -10

Reply:
how are you getting initial device ?

Reply:
omp_get_initial_device()